src/resources/espnow.cc

```diff
@@ -46,43 +49,67 @@ struct DataGram {
 
 const int kInvalidEspNow = -1;
 
+// These constants must be synchronized with the Toit code.
+const int kDataAvailableState = 1 << 0;
+const int kSendDoneState = 1 << 1;
+
+enum class EspNowEvent {
+  new_data_available = 1 << 0,
+  // Indicates that the sending has finished. Verify with 'status'
+  // that it was successful.
+  send_done = 1 << 1,
+};
+
 // Only allow one instance to use espnow.
 static  ResourcePool<int, kInvalidEspNow> espnow_pool(
   0
 );
 
-static SemaphoreHandle_t tx_sem;
 static esp_now_send_status_t tx_status;
```

Comment:
It feels like this could be in the resource, but maybe that is a bit hard to do.

Reply:
This variable is set from a callback (`espnow_send_cb`) and there is no way to get the resource in that function.